notebooks/exercise7-oqmd-optimade-tutorial.ipynb

```diff
@@ -0,0 +1,645 @@
+    "\n",
+    "$ git clone git@github.com:tachyontraveler/magpie_workflow.git\n",
+    "\n",
+    "> Cloning into 'magpie_workflow'...\n",
+    ">\n",
+    "> remote: Enumerating objects: 115, done.\n",
+    ">\n",
+    "> remote: Counting objects: 100% (6/6), done.\n",
+    ">\n",
+    "> remote: Compressing objects: 100% (6/6), done.\n",
+    ">\n",
+    "> remote: Total 115 (delta 2), reused 0 (delta 0), pack-reused 109\n",
+    ">\n",
+    "> Receiving objects: 100% (115/115), 16.03 MiB | 3.25 MiB/s, done.\n",
+    ">\n",
+    "> Resolving deltas: 100% (31/31), done.\n",
+    "\n",
+    "\n",
+    "$ cp -r ./input_poscars magpie_workflow/\n",
+    "\n",
+    "$ cd magpie_workflow\n",
+    "\n",
+    "$ python3 workflow.py \n",
+    "\n",
+    ">Initializing the workflow class\n",
+    ">\n",
+    "> 2021-11-22 19:51:45.614116 ::  Generating property.txt file\n",
+    ">\n",
+    "> 2021-11-22 19:51:45.616918 ::  Generating Magpie input commands file \n",
+    ">\n",
+    "> 2021-11-22 19:51:45.617025 ::  Magpie input file created as ./OUTDIR/generate-attributes.in\n",
+    ">\n",
+    "> 2021-11-22 19:51:45.617060 ::  Calling Magpie with the input script.. \n",
+    ">\n",
+    ">2021-11-22 19:51:45.617102 ::  May check out.workflow.txt for Magpie messages\n",
+    ">\n",
+    ">2021-11-22 19:52:36.347243 ::  Finished Magpie generation. Now post-processing\n",
+    ">\n",
+    ">2021-11-22 19:52:36.399678 ::  Done\n",
+    "\n",
+    "$ cd ../\n",
+    "\n",
+    "$ cp magpie_workflow/OUTDIR/features_processed_final.csv ./\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 3: Machine Learning\n",
+    "\n",
+    "#### Now we can proceed to build a small ML model using Scikit Learn library modules and fit it on the data obtained above"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the required libraries\n",
+    "\n",
+    "import numpy as np\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.pipeline import Pipeline \n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.decomposition import PCA\n",
+    "from sklearn.ensemble import RandomForestRegressor as RFR\n",
+    "from sklearn.svm import SVR\n",
+    "from sklearn.model_selection import GridSearchCV\n",
+    "import matplotlib.pyplot as plt "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the data generated in previous sections\n",
+    "\n",
+    "# Input feature data - generated using Magpie on POSCAR files \n",
+    "features = open(\"./features_processed_final.csv\",\"r\").read().strip().split(\"\\n\")\n",
```

Comment:
```suggestion
    "if os.path.isfile(\"./features_processed_final.csv\"): # Case where tthe optional part has been done\n",
    "    features = open(\"./features_processed_final.csv\",\"r\").read().strip().split(\"\\n\")\n",
    "else: \n",
    "    features = open(\"./oqmd_exercise_data/features_processed_final.csv\",\"r\").read().strip().split(\"\\n\") # When the optional part was not done\n",
```
If they have not done the optional part, the file: "features_processed_final.csv" is not present in the directory where the Jupyter notebook is stored, and they instead need to load it from the oqmd_exercise_data folder